### ISSUE TITLE
Undefined dynamic prop gives misleading error message

### ISSUE THREAD

**Issue by laem** [jantimon/next-yak#229]
```
//Component
return (						
<Group key={group} >...</Group> // missing $groupColor={groupColor}
)


const Group = styled.li`
	border-left: 4px solid ${(p) => p.$groupColor};
	padding-left: 0.4rem;
`


```  
![image](https://github.com/user-attachments/assets/f8d85f1a-dee9-4cfb-b6dc-5893f06f0bfc)

It took me a while to understand that the problem was an undefined prop. I was looking for some place in my code where CSS was written without being enclosed by the `css` template literal function. 

Solution : change the message for something like "<GroupColor styled component has a missing prop"

**Comment by jantimon** [comment]
I believe we don't know the Component name but we could at least show the css class name which failed to be generated

**Comment by jantimon** [comment]
I've improved the error message to help developers better find the reason for that error by including  the actual function that caused the issue

In your case that would have been:

```
Error: Dynamic CSS functions must return a string or number but returned undefined

Dynamic CSS function: (p) => p.$groupColor
```

![preview of the next version](https://github.com/user-attachments/assets/dd941465-38f7-4ecd-b52f-260326aa84d0)

Would that have been helpful for you?

**Comment by laem** [comment]
Yes, vert helpful !

**Comment by jantimon** [comment]
published in 3.1.0
